docs/rebuilding.md

```diff
@@ -0,0 +1,70 @@
+# Rebuilding Packages & Creating the ConnectBox repository
+
+Due to architectural requirements of some models of the Raspberry Pi, it is necessary to ocassionally rebuild some of the packages used by the ConnectBox (see [Raspberry_Pi_issues](https://wiki.debian.org/RaspberryPi#Raspberry_Pi_issues) for details) and host them in our own repository.
+
+Complete documentation on how to build a package for Debian can be found at their [BuildingTutorial](https://wiki.debian.org/BuildingTutorial), so the steps will only be summarized here.
+
+## Install the toolchain
+
+The easiest way to build compatible packages is directly on the Raspberry Pi 3 (The alternative is going through the effort of setting up a cross compiler toolchain on faster hardware).
+
+    sudo apt-get install build-essential fakeroot devscripts
+
+## Set up a source repository
+
+The robust Debian ecosystem allows us to reproducibly rebuild almost any package without significant effort. One aspect of this is that every major package repository has a cooresponding source repository. For these instructions, add the jessie-backports source repo to apt:
+
+    echo 'deb-src http://ftp.debian.org/debian jessie-backports main contrib' > /etc/apt/sources.list.d/jessie-backports-src.list
+
+## Rebuild the package
+
+Make a directory to work in (the name is not important)
+
+    mkdir debian
+    cd debian
+
+Get the source code from the src repo
+
+    apt-get apt-get source libssl-dev
+
+Get the build dependecies (Note: you may run into dependency hell here, since often the reason for rebuilding a package is to get a newer version... which itself requires new libraries... the only way out is to follow the depedencies down, building them too)
+
+    sudo apt-get build-dep libssl-dev
+
+Rebuild the package
+
+    cd openssl-1.0.2k
+    debuild -b -uc -us
+
+If all goes well the system will chug for a while and you will have several freshly built packages in your work directory (Somewhat counterituitively, they land outside the directory you ran the debuild command in).
```

Comment:
The system chugged for a while, but the package didn’t build cleanly. It finished with

```
…
( :; LIBDEPS="${LIBDEPS:--L.. -lssl -L.. -lcrypto  -ldl}"; LDCMD="${LDCMD:-gcc}"; LDFLAGS="${LDFLAGS:--DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H -DL_ENDIAN -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -Wl,-z,relro -Wa,--noexecstack -Wall -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DAES_ASM -DBSAES_ASM -DGHASH_ASM}"; LIBPATH=`for x in $LIBDEPS; do echo $x; done | sed -e 's/^ *-L//;t' -e d | uniq`; LIBPATH=`echo $LIBPATH | sed -e 's/ /:/g'`; LD_LIBRARY_PATH=$LIBPATH:$LD_LIBRARY_PATH ${LDCMD} ${LDFLAGS} -o ${APPNAME:=dtlstest} dtlstest.o ssltestlib.o ${LIBDEPS} )
collect2: error: ld returned 1 exit status
../Makefile.shared:164: recipe for target 'link_app.' failed
make[3]: *** [link_app.] Error 1
make[3]: Leaving directory '/home/pi/debian/openssl-1.0.2k/test'
Makefile:567: recipe for target 'dtlstest' failed
make[2]: *** [dtlstest] Error 2
make[2]: Leaving directory '/home/pi/debian/openssl-1.0.2k/test'
Makefile:296: recipe for target 'build_tests' failed
make[1]: *** [build_tests] Error 1
make[1]: Leaving directory '/home/pi/debian/openssl-1.0.2k'
debian/rules:49: recipe for target 'build-stamp' failed
make: *** [build-stamp] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debuild: fatal error at line 1376:
dpkg-buildpackage -rfakeroot -D -us -uc -b failed
```

I’m not sure whether that’s because there’s a step missing, or just a problem with the upstream package itself.